sync3/handler/connstate_live.go

```diff
@@ -284,16 +284,34 @@ func (s *connStateLive) processUpdatesForSubscriptions(ctx context.Context, buil
 
 // this function does any updates which apply to the connection, regardless of which lists/subs exist.
 func (s *connStateLive) processGlobalUpdates(ctx context.Context, builder *RoomsBuilder, up caches.Update) (delta sync3.RoomDelta) {
-	rup, ok := up.(caches.RoomUpdate)
-	if ok {
+	bumpThisRoom := true
+
+	roomEventUpdate, isRoomEventUpdate := up.(*caches.RoomEventUpdate)
+	if isRoomEventUpdate {
+		// If this connection cares about specific event types, only bump the room
+		// for room updates that the connection cares about.
+		// TODO: this probably ought to be in SetRoom, not in here. But SetRoom doesn't
+		// have access to muxedReq, where BumpEventTypes lives (for now).
```

Comment:
This is pretty ugly. I guess I could change how `s.lists` is managed so that it tracks a copy of the BumpEventTypes. Not sure I want to go down that route unless we're happy to commit to `bump_event_types` being per-list; that'd mean something like a per-list LastMessageTimestamp or a per event-type LastMessageTimestamp across rooms.

Reply:
`replacePreviousTimestamp` in 7feabcd.